server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1259,18 +1247,71 @@ boolean canSuspendTablets() {
             // setting.
             return getConfiguration().getBoolean(Property.MANAGER_METADATA_SUSPENDABLE);
           }
-        });
+        };
+    watchers.add(metadataTableTGW);
 
-    watchers
-        .add(new TabletGroupWatcher(this, this.rootTabletStore, watchers.get(1), managerMetrics) {
+    final TabletGroupWatcher rootTableTGW =
+        new TabletGroupWatcher(this, this.rootTabletStore, watchers.get(1), managerMetrics) {
           @Override
           boolean canSuspendTablets() {
             // Never allow root tablet to enter suspended state.
             return false;
           }
-        });
-    for (TabletGroupWatcher watcher : watchers) {
-      watcher.start();
+        };
+    watchers.add(rootTableTGW);
+
+    while (isUpgrading()) {
```

Comment:
Wonder if some logging in this loop would be useful, depends on what is currently be logged on by other code like TGW.

Reply:
`UpgradeCoordinator.setStatus ` calls the EventCoordinator when the status of the upgrade changes. These messages get logged.